packages/webapp/pages/squads/[handle].tsx

```diff
@@ -0,0 +1,134 @@
+import { useRouter } from 'next/router';
+import { NextSeoProps } from 'next-seo/lib/types';
+import { NextSeo } from 'next-seo';
+import Feed from '@dailydotdev/shared/src/components/Feed';
+import { SOURCE_FEED_QUERY } from '@dailydotdev/shared/src/graphql/feed';
+import AuthContext from '@dailydotdev/shared/src/contexts/AuthContext';
+import { SquadPageHeader } from '@dailydotdev/shared/src/components/squads/SquadPageHeader';
+import { FeedPage } from '@dailydotdev/shared/src/components/utilities';
+import {
+  getSquad,
+  getSquadMembers,
+  Squad,
+  SquadMember,
+} from '@dailydotdev/shared/src/graphql/squads';
+import Custom404 from '../404';
+import { defaultOpenGraph, defaultSeo } from '../../next-seo';
+import { mainFeedLayoutProps } from '../../components/layouts/MainFeedPage';
+import { getLayout } from '../../components/layouts/FeedLayout';
+
+type SourcePageProps = {
+  squad: Squad;
+  squadMembers: SquadMember[];
+  squadMemberCount: number;
+};
+
+const SquadPage = ({
+  squad,
+  squadMembers,
+  squadMemberCount,
+}: SourcePageProps): ReactElement => {
+  const { isFallback } = useRouter();
+  const { user } = useContext(AuthContext);
+  // Must be memoized to prevent refreshing the feed
+  const queryVariables = useMemo(
+    () => ({ source: squad?.id, ranking: 'TIME' }),
+    [squad?.id],
+  );
+
+  if (!isFallback && !squad) {
+    return <Custom404 />;
+  }
+
+  if (isFallback || !squad) {
+    return <></>;
+  }
+
+  const seo: NextSeoProps = {
+    title: `${squad.name} posts on daily.dev`,
+    openGraph: { ...defaultOpenGraph },
+    ...defaultSeo,
+  };
+
+  return (
+    <FeedPage
+      className="laptop:pr-0 laptop:pl-0 mb-4"
+      style={{
+        background: 'radial-gradient(ellipse, #c029f088 0%, #c029f000 400px)',
+        backgroundSize: '1200px 500px',
+        backgroundPosition: 'center -270px',
+        backgroundRepeat: 'no-repeat',
+      }}
+    >
+      <NextSeo {...seo} />
+      <SquadPageHeader
+        squad={squad}
+        members={squadMembers}
+        memberCount={squadMemberCount}
+      />
+      <Feed
+        feedName="source"
+        feedQueryKey={[
+          'sourceFeed',
+          user?.id ?? 'anonymous',
+          Object.values(queryVariables),
+        ]}
+        query={SOURCE_FEED_QUERY}
+        variables={queryVariables}
+      />
+    </FeedPage>
+  );
+};
+
+SquadPage.getLayout = getLayout;
+SquadPage.layoutProps = mainFeedLayoutProps;
+
+export default SquadPage;
+
+export async function getStaticPaths(): Promise<GetStaticPathsResult> {
+  return { paths: [], fallback: true };
+}
+
+interface SquadPageParams extends ParsedUrlQuery {
+  handle: string;
+}
+
+export async function getStaticProps({
+  params,
+}: GetStaticPropsContext<SquadPageParams>): Promise<
+  GetStaticPropsResult<SourcePageProps>
+> {
+  try {
+    const squad = await getSquad(params.handle);
+    const squadMembers = await getSquadMembers(squad.id);
```

Comment:
Is this the only implementation of these functions?
If so maybe we can simply call the functions directly here.

Reply:
Yes for now, but it may be used later since we have a lot to do